src/main/java/games/strategy/triplea/ui/MovePanel.java

```diff
@@ -1346,6 +1346,7 @@ protected final void cleanUpSpecific() {
     selectedUnits.clear();
     updateRouteAndMouseShadowUnits(null);
     forced = null;
+    getMap().showMouseCursor();
```

Comment:
Nope, this code is new (AFAICT going back to 2007 when `MapPanel#showMouseCursor()` was added).  This is my solution to the brittleness you mentioned above.  See my [previous comment](https://github.com/triplea-game/triplea/pull/2186#issuecomment-320782576) for an explanation.

Basically, now there may potentially be two calls to `getMap().showMouseCursor()` at the completion of a move from two different code paths.  But it should be less likely for a dev to get burned in the future by the roundabout nature of one of those paths that originate in `AbstractMovePanel#setActive()`.

I don't mind removing it if you think it just muddies the waters even more.  Its presence or absence will not affect the fix for #2185.